kernel/src/log_segment.rs

```diff
@@ -166,21 +166,59 @@ impl LogSegment {
                 .filter_ok(|x| x.is_commit())
                 .try_collect()?;
 
+        // Return a FileNotFound error if there are no new commits.
+        // Callers can then decide how to handle this case.
+        require!(
+            !ascending_commit_files.is_empty(),
+            Error::file_not_found("No new commits.")
```

Comment:
I think the error should be more like "No commits in range", or "Failed to find commits in the range".  "New commits" makes sense for an update operation, but CDF could operate over old commits.